packages/roosterjs-content-model-core/lib/coreApi/setDOMSelection/setDOMSelection.ts

```diff
@@ -41,14 +46,19 @@ export const setDOMSelection: SetDOMSelection = (core, selection, skipSelectionC
             case 'image':
                 const image = selection.image;
 
+                const spanImage = ensureImageHasSpanParent(image);
+                if (!spanImage) {
```

Comment:
This should never be true. You can declare the function `ensureImageHasSpanParent` to let it return HTMLElement so no need to check here